src/lib/cache/hash_cache.hpp

```diff
@@ -9,22 +9,20 @@
 
 #include "gdfs_cache.hpp"
 
-#include "SQLParserResult.h"
 #include "utils/singleton.hpp"
 
 namespace opossum {
 
 inline constexpr size_t DefaultCacheCapacity = 1024;
 
-// Cache that stores instances of SQLParserResult.
 // Per-default, uses the GDFS cache as underlying storage.
 template <typename Value, typename Key = std::string>
-class SQLQueryCache : public Singleton<SQLQueryCache<Value, Key>> {
+class HashCache : public Singleton<HashCache<Value, Key>> {
```

Comment:
`HashBasedCache`?

Reply:
I don't like it, I think "Based" is vague, implies there is something deeper going on, makes the reader unsure. "HashCache" says "it's a cache using hashed keys, nothing special, nothing to worry about".  

I think the term "HashCache" is in line with terms like "HashMap", "HashSet", "SingleLinkedList", "DoubleLinkedList" - none of these need a "Based" in their name.

Reply:
Can you think of a type of cache that doesn't use hashes? Maybe "Cache" is sufficient?

Reply:
Well, the other alternative would be to rely on ordering instead of hashing (see map vs unordered_map). 
Your call, I like "HashCache" best, "Cache" second best, and "HashBasedCache" the least.

Reply:
I am not sure if I have seen equi-based caching in the wild.

Looking at hash_cache.hpp, we don't even have anything that limits us to hash-based caching:

```
 bool has(const Key& query) const { return _cache->has(query); }
```

I would go for Cache here, but just saw a second issue: In the cache, we have a pointer to an `AbstractCache`. No matter what we call this thing, this seems inconsistent. *Impl?

Reply:
`Cache` it is then; and `_impl`

Reply:
I was also thinking about the class names themselves. It seems weird to have a `GDFSCache,` which should not be used as an interface to a cache, and a `Cache` that is not an `AbstractCache`.

Reply:
Renamed it to AbstractCacheEvictionPolicy - still not clean, because the derived types also handle the storage, but still better than the current naming.

Reply:
Why not `AbstractCacheImpl` (which I meant with *Impl above) or `AbstractCacheBackend`?

Reply:
AbstractCacheImpl, done